Formula/sapling.rb

```diff
@@ -35,7 +35,7 @@ def install
     python3 = "python3.11"
 
     ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
-    ENV["SAPLING_VERSION"] = version.to_s
+    ENV["SAPLING_VERSION"] = version.to_s.split("-").take(2).join("-")
```

Comment:
This is one way work around the failure. This was the version format we used before #126198 (https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+is%3Aclosed+sapling).

But on second thoughts, this also means that the version reported by the formula will be different from the one output by `sl --version`; the last part of the version string `-h69692651` is removed by this workaround. (Actually, the inconsistency has already been the case since #126198.) So the test below may not be happy about it.

We could use this workaround for the test below too. But I do think that we should make the version number consistent here.

Cc @samford for thoughts, since you authored #126198.

Reply:
You also need to update the test.
```
  ==> /opt/homebrew/Cellar/sapling/0.2.20230330-193452-h69692651/bin/sl --version
  Error: sapling: failed
  An exception occurred within a child process:
    Minitest::Assertion: --- expected
  +++ actual
  @@ -1 +1 @@
  -"Sapling 0.2.20230330-193452-h69692651"
  +"Sapling 0.2.20230330-193452"
```